ocaml/xapi-cli-server/records.ml

```diff
@@ -5242,3 +5242,119 @@ let vtpm_record rpc session_id vtpm =
           ()
       ]
   }
+
+let tracing_record rpc session_id tracing =
+  let _ref = ref tracing in
+  let empty_record =
+    ToGet (fun () -> Client.Tracing.get_record ~rpc ~session_id ~self:!_ref)
+  in
+  let record = ref empty_record in
+  let x () = lzy_get record in
+  {
+    setref=
+      (fun r ->
+        _ref := r ;
+        record := empty_record
+      )
+  ; setrefrec=
+      (fun (a, b) ->
+        _ref := a ;
+        record := Got b
+      )
+  ; record= x
+  ; getref= (fun () -> !_ref)
+  ; fields=
+      [
+        make_field ~name:"uuid" ~get:(fun () -> (x ()).API.tracing_uuid) ()
+      ; make_field ~name:"hosts"
+          ~get:(fun () ->
+            List.map get_uuid_from_ref (x ()).API.tracing_hosts
+            |> String.concat ","
+            |> fun s -> if s = "" then "all" else ""
+          )
+          ()
+      ; make_field ~name:"name_label"
+          ~get:(fun () -> (x ()).API.tracing_name_label)
+          ()
+      ; make_field ~name:"tags"
+          ~get:(fun () ->
+            List.map
+              (fun (k, v) -> Printf.sprintf "%s : %s" k v)
+              (x ()).API.tracing_tags
+            |> String.concat ","
+            |> fun s -> if s = "" then "none" else s
```

Comment:
This doesn't follow the convention elsewhere. Simply use the empty list when there are no values

Reply:
I can do that, it makes sense if it is converntion